api/v1/serializers/instance_serializer.py

```diff
@@ -51,6 +52,14 @@ def __init__(self, *args, **kwargs):
         self.request_user = user
         super(InstanceSerializer, self).__init__(*args, **kwargs)
 
+    def get_size(self, obj):
```

Comment:
I can't find where this new function comes into play.  Does it get called in some way by your frontend code?